tests/testapp/tests/test_syncsession_methods.py

```diff
@@ -0,0 +1,251 @@
+        self.assertEqual(rmc.counter, rmcb.counter)
+
+    def test_dequeuing_merge_conflict_buffer(self):
+        store = Store.objects.get(id=self.data['model2'])
+        self.assertNotEqual(store.last_saved_instance, self.current_id.id)
+        self.assertEqual(store.conflicting_serialized_data, "store")
+        with connection.cursor() as cursor:
+            current_id = InstanceIDModel.get_and_update_current_instance()
+            self.data['sc']._dequeuing_merge_conflict_buffer(cursor, current_id)
+        store = Store.objects.get(id=self.data['model2'])
+        self.assertEqual(store.last_saved_instance, current_id.id)
+        self.assertEqual(store.last_saved_counter, current_id.counter)
+        self.assertEqual(store.conflicting_serialized_data, "buffer\nstore")
+
+    def test_dequeuing_update_rmcs_last_saved_by(self):
+        self.assertFalse(RecordMaxCounter.objects.filter(instance_id=self.current_id.id).exists())
+        with connection.cursor() as cursor:
+            current_id = InstanceIDModel.get_and_update_current_instance()
+            self.data['sc']._dequeuing_update_rmcs_last_saved_by(cursor, current_id)
+        self.assertTrue(RecordMaxCounter.objects.filter(instance_id=current_id.id).exists())
+
+    def test_dequeuing_delete_mc_buffer(self):
+        self.assertTrue(Buffer.objects.filter(model_uuid=self.data['model2']).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_delete_mc_buffer(cursor)
+        self.assertFalse(Buffer.objects.filter(model_uuid=self.data['model2']).exists())
+
+    def test_dequeuing_delete_mc_rmcb(self):
+        self.assertTrue(RecordMaxCounterBuffer.objects.filter(model_uuid=self.data['model2'], instance_id=self.data['model2_rmcb_ids'][0]).exists())
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_delete_mc_rmcb(cursor)
+        self.assertFalse(RecordMaxCounterBuffer.objects.filter(model_uuid=self.data['model2'], instance_id=self.data['model2_rmcb_ids'][0]).exists())
+
+    def test_dequeuing_ff_rmc(self):
+        # this only transfers rmcs that have the same instance id (does not handle instance ids it has not seen before)
+        rmc = RecordMaxCounter.objects.get(instance_id=self.data['model3_rmc_ids'][0], store_model_id=self.data['model3'])
+        rmcb = RecordMaxCounterBuffer.objects.get(instance_id=self.data['model3_rmcb_ids'][0], model_uuid=self.data['model3'])
+        self.assertNotEqual(rmc.counter, rmcb.counter)
+        with connection.cursor() as cursor:
+            self.data['sc']._dequeuing_ff_rmc(cursor)
+        rmc = RecordMaxCounter.objects.get(instance_id=self.data['model3_rmc_ids'][0], store_model_id=self.data['model3'])
+        rmcb = RecordMaxCounterBuffer.objects.get(instance_id=self.data['model3_rmcb_ids'][0], model_uuid=self.data['model3'])
+        self.assertEqual(rmc.counter, rmcb.counter)
```

Comment:
Deleted this test since we are not using it, but since we just push all remaining records into the store (after all the deletions), then those instance IDs that did not exist before are now in the RMC.